duckdb-vortex/src/vortex_scan.cpp

```diff
@@ -231,6 +239,12 @@ static unique_ptr<FileReader> OpenFileAndVerify(FileSystem &fs, const std::strin
 	return file;
 }
 
+static bool PinFileToThread(ScanGlobalState &global_state) {
+	const auto thread_count = std::thread::hardware_concurrency();
+	const auto file_count = global_state.expanded_files.size();
+	return (file_count - global_state.files_partitioned) > thread_count;
+}
```

Comment:
`thread_count` is likely much larger than the number of threads working on this scan.

Reply:
As discussed offline, for now this is good enough as in an approximation that if wrong might be a small perf hit.

Reply:
Added a comment